### PR TITLE
Update configs for lower envs

### PR DESCRIPTION
**Pull request by Shakespeared**
ticket: https://broadworkbench.atlassian.net/browse/ID-546

This adds DRSHub config blocks to specify host URIs in specific lower environments (specifically alpha and staging). The configs should match the configs in Martha in functionality, although they are organized differently. It should map to the Martha configs (found [here](https://github.com/broadinstitute/martha/blob/dev/common/config.js)). 

The main change in this PR is to explicitly set the TDR URI (expanded from the new anvil CIB which is drs.anv0) to different values in dev, staging, and alpha. 

I went back and forth a bit on the best way to structure the config blocks, I think I landed on something that makes sense, but if this doesn't look right please let me know.

### CODE REVIEW

**Review by Ghost-in-a-Jar — APPROVED** [review]

File: service/src/main/resources/application.yml [hunk 1]
```diff
@@ -132,4 +125,42 @@ drshub:
       mTlsConfig:
         keyPath: rendered/ras-mtls-client.key
         certPath: rendered/ras-mtls-client.crt
+  compactIdHosts:
+    "dg.4503": staging.gen3.biodatacatalyst.nhlbi.nih.gov
```
Comment (mbaumann-broad) [inline service/src/main/resources/application.yml]: `dg.4503` is the BDC compact id for production only.
It should never appear or be used in preprod tiers. 
Cross-deployment-tier data access does not work in general (for Gen3 or TDR)

If this line were deleted, and DRSHub unexpectedly received a DRS URI containing `dg.4503` would this result in the "compact identifier not found" error?
If so, then I think removing this line (or replacing it with a comment) is the appropriate change.

File: service/src/main/resources/application.yml [hunk 2]
```diff
@@ -132,4 +125,42 @@ drshub:
...
+    "dg.4dfc": nci-crdc-staging.datacommons.io
+    "dg.f82a1a": gen3staging.kidsfirstdrc.org
+---
+# prod
+spring.config.activate.on-profile: prod
+drshub:
+  compactIdHosts:
+    "dg.4503": gen3.biodatacatalyst.nhlbi.nih.gov # biodatacatalyst
+    # this compact ID intentionally maps to the BDC staging url in prod
+    "dg.712c": staging.gen3.biodatacatalyst.nhlbi.nih.gov # biodatacatalyst
```
Comment (mbaumann-broad) [inline service/src/main/resources/application.yml]: `dg.712c` is the BDC compact id for preproduction only.
It should never appear or be used in the production tier. 
Cross-deployment-tier data access does not work in general (for Gen3 or TDR)

If this line were deleted, and DRSHub unexpectedly received a DRS URI containing `dg.712c` would this result in the "compact identifier not found" error?
If so, then I think removing this line (or replacing it with a comment) is the appropriate change.

**Review by mbaumann-broad — COMMENTED** [review]
Thank you for these changes @Shakespeared!
Overall, the changes look good to me.
It will be great having DRSHub working as expected in the Alpha and Staging tiers, which currently include tests for Martha.

I left two comments regarding specific corner cases for BDC.
Please review these and post your thoughts. Thank you!

File: service/src/main/resources/application.yml (hunk 1 above)
Comment (Shakespeared) [reply service/src/main/resources/application.yml]: Great info, thanks for surfacing this! You're correct in that if this is removed from specific tiers, it should throw the "compact identifier not found" error if a URL with this id is resolved. That sounds like the correct behavior to me, I'll update the configs

File: service/src/main/resources/application.yml [hunk 3]
```diff
@@ -7,7 +7,6 @@ drshub:
 
   compactIdHosts:
     # it is important for case-insensitive matching that the keys here are all lower case
-    "dg.4503": gen3.biodatacatalyst.nhlbi.nih.gov
```
Comment (mbaumann-broad) [inline service/src/main/resources/application.yml:10]: The "prod" and "!prod" definitions for "dg.4503" and "dg.712c" later in this file look good and make sense.

I am less clear about when the definitions in this section would come into play.
Depending on when/how the values in this section are used, my general sense is that either:
- Both "dg.4503" and "dg.712c" would remain here.   
- or, both "dg.4503" and "dg.712c" would be removed here. 

Please advise. :)

Comment (Shakespeared) [reply service/src/main/resources/application.yml:10]: I had a look at the docs, the default case will appear in all envs unless explicitly overwritten, so I removed both of the env-specific CIBs from here, so each should only populate in the correct env now. Thanks for catching that!

Comment (mbaumann-broad) [reply service/src/main/resources/application.yml:10]: For the same reason, these can(/should) also be removed from the common block, as these all correctly appear in the `prod` and `!prod` blocks below:
```
    "dg.anv0": staging.theanvil.io
    "drs.anv0": jade.datarepo-dev.broadinstitute.org
    "dg.4dfc": nci-crdc-staging.datacommons.io
    "dg.f82a1a": gen3staging.kidsfirstdrc.org
  ```

**Review by mbaumann-broad — CHANGES_REQUESTED** [review]
Thank you, @Shakespeared!
I think there is the opportunity for a little more cleanup along the same lines and for the same reasons.

**Review by mbaumann-broad — APPROVED** [review]
LGTM!
Thank you for the work on this, @Shakespeared!
